Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -361,7 +368,7 @@ static inline void GeneratePixelShader(T& out, DSTALPHA_MODE dstAlphaMode, API_T
 		out.Write("void main(\n");
 		out.Write("  out float4 ocol0 : SV_Target0,%s%s\n  in float4 rawpos : SV_Position,\n",
 			dstAlphaMode == DSTALPHA_DUAL_SOURCE_BLEND ? "\n  out float4 ocol1 : SV_Target1," : "",
-			per_pixel_depth ? "\n  out float depth : SV_Depth," : "");
+			(per_pixel_depth && bpmem.zmode.testenable) ? "\n  out float depth : SV_Depth," : "");
```

Comment:
why bpmem.zmode.testenabled is not directly included as a condition for per_pixel_depth? shure this code works as intended because if a shader is compiled with per pixel depth enabled but depth test disabled a reference to depth variable will be used but not declared